### PR TITLE
Device support for the Xiaomi Mi Smart Fan

### PR DESCRIPTION
**Pull request by syssi**

### CODE REVIEW

File: mirobo/fan.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+    def set_natural_level(self, level: int):
+        """Set natural level."""
+        return self.send("set_natural_level", [level])
+
+    def set_speed_level(self, level: int):
+        """Set speed level."""
+        return self.send("set_speed_level", [level])
+
+    def set_move(self, direction: MoveDirection):
```
Comment (rytilahti) [inline mirobo/fan.py]: set_direction maybe?

File: mirobo/fan.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        """Power off."""
+        return self.send("set_power", ["off"])
+
+    def set_natural_level(self, level: int):
+        """Set natural level."""
+        return self.send("set_natural_level", [level])
+
+    def set_speed_level(self, level: int):
+        """Set speed level."""
+        return self.send("set_speed_level", [level])
```
Comment (rytilahti) [inline mirobo/fan.py]: What are natural and speed levels?

File: mirobo/fan.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        return self.send("set_angle_enable", ["on"])
+
+    def oscillate_off(self):
+        """Disable oscillate."""
+        return self.send("set_angle_enable", ["off"])
+
+    def set_led_brightness(self, brightness: LedBrightness):
+        """Set led brightness."""
+
+        # bright: 0, dim: 1, off: 2
```
Comment (rytilahti) [inline mirobo/fan.py]: This is already encoded in the enum, so please remove the comment.

**Comment by syssi** [comment]
Hmm... it looks like I'm still unable to do a proper rebase. :-(

**Comment by rytilahti** [comment]
The following works for me usually, I don't do pull but use fetch and rebase instead of it like so:
```
git fetch upstream
git rebase upstream/master
```

**Comment by syssi** [comment]
Hmm..  I did the same, fixed the conflict, was forced to "add" and "git rebase --continue". To push the result I was forced to do a merge commit. I'll be damned!

**Comment by rytilahti** [comment]
Maybe it would make just sense to create a new branch on top of the current master, and cherry-pick the commits you want from this one there? That would keep the history clean.

**Pull request closed without merging**